### PR TITLE
feat: [PL-29721]: disabled msg on gcp SM

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary
Showing a message why few fields are not editable in GCP SM secrets

#### Screenshots
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 1]
```diff
@@ -499,15 +501,23 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
...
-                idName="identifier"
-                isIdentifierEditable={!editing}
-                inputGroupProps={{
-                  disabled: isGcpSMInlineEditMode() || loading
-                }}
-              />
+              <Popover
+                interactionKind={'hover-target'}
+                position="top"
+                className={css.hoverMsg}
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:507]: do we need display grid?

File: src/modules/30-secrets/components/CreateUpdateSecret/views/VaultFormFields.tsx [hunk 2]
```diff
@@ -95,21 +103,30 @@ const VaultFormFields: React.FC<VaultFormFieldsProps & FormikContextProps<any>>
...
-                items={regions}
+          <Popover
+            interactionKind={'hover-target'}
+            position="top"
+            className={css.hoverMsg}
+            content={<Text padding="medium">{getString('secrets.gcpSecretEdit')}</Text>}
+            disabled={!gcpSmInEditMode()}
+          >
+            <>
+              <FormInput.CheckBox
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/views/VaultFormFields.tsx:114]: I think we already have the support of tooltipProps in Form Inputs. Can you check once?

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/30-secrets/components/CreateUpdateSecret/views/VaultFormFields.tsx (hunk 2 above)
Comment (swarajcheguri-harness) [reply src/modules/30-secrets/components/CreateUpdateSecret/views/VaultFormFields.tsx:116]: no we dont support tooltip message

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx (hunk 1 above)
Comment (swarajcheguri-harness) [reply src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:507]: corrected

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**